dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+        break;
+      case 'Z':
+        dartCommandClass = '_PathClose';
+        break;
+      default:
+        throw new Exception('unsupported path command: $type');
+    }
+    String result = '${kIndent*4}const $dartCommandClass(\n';
+    for (List<Point<double>> pointFrames in points) {
+      result += '${kIndent*5}const <Offset> [\n';
+      for (Point<double> point in pointFrames)
+        result += '${kIndent*6}const Offset(${point.x}, ${point.y}),\n';
+      result += '${kIndent*5}],\n';
+    }
+    result +='${kIndent*4}),\n';
+    return result;
+  }
+}
+
+/// Interprets some subset of an SVG* file.
+///
+/// Recursively goes over the SVG tree, applying transforms and opacities,
+/// and build a FrameData which is a flat representation of the paths in the SVG
+/// file, after applying transformations and converting relative coordinates to
+/// absolute.
+///
+/// * Note that this does not support the SVG specification, but is just built to
+/// support SVG files exported by a specific tool the motion design team is
+/// using.
+FrameData interpretSvg(String svgFilePath) {
+  final File file = new File(svgFilePath);
+  if (!file.existsSync()) {
+    throw new ArgumentError('$file does not exist');
+  }
+  final String fileData = file.readAsStringSync();
+  final XmlElement svgElement = _extractSvgElement(xml.parse(fileData));
+  final double width = parsePixels(_extractAttr(svgElement, 'width')).toDouble();
+  final double height = parsePixels(_extractAttr(svgElement, 'height')).toDouble();
+
+  final List<SvgPath> paths =
+    _interpretSvgGroup(svgElement.children, new _Transform());
+  return new FrameData(new Point<double>(width, height), paths);
+}
+
+List<SvgPath> _interpretSvgGroup(List<XmlNode> children, _Transform transform) {
+  final List<SvgPath> paths = <SvgPath>[];
+  for (XmlNode node in children) {
+    if (node.nodeType != XmlNodeType.ELEMENT)
+      continue;
+    final XmlElement element = node;
+
+    if (element.name.local == 'path') {
+      // TODO(amirh): convert relative commands to absolute
+      paths.add(SvgPath.fromElement(element).applyTransform(transform));
+    }
+
+    if (element.name.local == 'g') {
+      double opacity = transform.opacity;
+      if (_hasAttr(element, 'opacity'))
+        opacity *= double.parse(_extractAttr(element, 'opacity'));
+
+      Matrix3 transformMatrix = transform.transformMatrix;
+      if (_hasAttr(element, 'transform'))
+        transformMatrix = transformMatrix.multiplied(
+            _parseSvgTransform(_extractAttr(element, 'transform')));
+
+      final _Transform subtreeTransform = new _Transform(
+        transformMatrix: transformMatrix,
+        opacity: opacity
+      );
+      paths.addAll(_interpretSvgGroup(element.children, subtreeTransform));
+    }
+  }
+  return paths;
+}
+
+// Given a points list in the form e.g: "25.0, 1.0 12.0, 12.0 23.0, 9.0" matches
+// the coordinated of the first point and the rest of the string, for the
+// example above:
```

Comment:
I'm pretty sure that SVG doesn't care if the commas are present or not, so this seems dubious

Reply:
made commas optional